scripts/logs-backup-from-all-pods/extract_logs.py

```diff
@@ -0,0 +1,187 @@
+import datetime
+import logging
+import subprocess
+import sys
+import json
+
+import psycopg2
+import redis
+import requests
+from datetime import datetime, timedelta
+
+def login_to_isd():
+    try:
+        cookie = ""
+        cmd = "curl -vvv -X POST '" + isd_gate_url + "/login?username=" + isd_admin_username + "&password=" + isd_admin_password + "&submit=Login'"
+        output = subprocess.getoutput(cmd=cmd)
+        output = output.replace(isd_admin_username, "***").replace(isd_admin_password, "***")
+        logging.info(f"Output for ISD login : {output}")
+        components = output.split("<")
+        for comp in components:
+            if comp.__contains__("set-cookie") or comp.__contains__("Set-Cookie"):
+                cookie = comp.split(":")[1].strip()
+        return cookie
+    except Exception as e:
+        print("Exception occurred while logging in to ISD : ", e)
+        logging.error("Exception occurred while logging in to ISD : ", exc_info=True)
+        raise e
+
+
+def get_response(url, cookie):
+    try:
+        logging.info(url)
+        headers = {'cookie': cookie, 'x-spinnaker-user': isd_admin_username}
```

Comment:
if cookie is passed then x-spinnaker-user is not required. GATE service will extract the username from the cookie

Reply:
Sending it just to be on the safe side.

I am not sure what was wrong but at places there was this header missing.

Reply:
thats not required. pls remove the header x-spinnaker-user